### PR TITLE
linuxPackages_latest.perf: apply upstream patch for binutils-2.39

### PR DESCRIPTION
**Pull request by trofi**
Without the change build fails as:

    perf-linux> util/annotate.c: In function 'symbol__disassemble_bpf':
    perf-linux> util/annotate.c:1765:9: error: too few arguments to function 'init_disassemble_info'
    perf-linux>  1765 |         init_disassemble_info(&info, s,
    perf-linux>       |         ^~~~~~~~~~~~~~~~~~~~~

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/perf.nix [hunk 1]
```diff
@@ -15,6 +15,13 @@ stdenv.mkDerivation {
 
   inherit (kernel) src;
```
Comment (amjoseph-nixpkgs) [inline pkgs/os-specific/linux/kernel/perf.nix:17]:
```suggestion

assert !(versionAtLeast kernel.version "5.19") -> throw "perf with binutils >= 2.39 requires kernel >= 5.19";

Comment (trofi) [reply pkgs/os-specific/linux/kernel/perf.nix:17]: I don't understand this assert. It does not mention binutils version in the constraints. Where does that requirement come from?

Also note that `master` branch does not yet have `binutils-2.39`.

Comment (amjoseph-nixpkgs) [reply pkgs/os-specific/linux/kernel/perf.nix:17]: > I don't understand this assert. It does not mention binutils version in the constraints. Where does that requirement come from?

I am suggesting that you include this change in your PR #185297, which bumps the binutils version to 2.39.

Github won't let me ```suggest this change in that PR.  Apparently I can only ```suggest changes within three lines of something the PR already touches.

Comment (trofi) [reply pkgs/os-specific/linux/kernel/perf.nix:17]: Ahh, that makes sense.

**Pull request merged**